src/pybind/mgr/volumes/fs/operations/versions/subvolume_v1.py

```diff
@@ -768,7 +768,16 @@ def get_pending_clones(self, snapname):
 
         for track_id in pending_track_id_list:
             try:
-                link_path = self.fs.readlink(os.path.join(index_path, track_id), 4096)
+                path = os.path.join(index_path, track_id)
+                link_path = self.fs.readlink(path, 4096)
+                try:
+                    self.fs.stat(link_path)
+                except cephfs.Error as e:
+                    if e.args[0] == errno.ENOENT:
+                        self.fs.unlink(path)
+                        self._remove_snap_clone(track_id)
+                    else:
+                        log.warning(f"Unlink operation could not be processed with the exception: {e}")
```

Comment:
Also add a info message before cleaning the dangling symlink for clone.

Suggestion:
```
try:
    self.fs.stat(link_path)
except cephfs.Error as e:
    if e.args[0] == errno.ENOENT:
        try:
            log.info(f"Clean up dangling symlink for the clone: {path})
            self.fs.unlink(path)
            self._remove_snap_clone(track_id)
        except (cephfs.Error, MetadataMgrException) as e:
            log.warning(f"Clean up of dangling symlink for the clone failed during unlink with the exception: {e}")
            pass
   
```

You can as well write a supporting routine (```def clean_dangling_clone_symlink```) to make it look cleaner it you think this is polluting the partent function